### PR TITLE
Adapters compatibility for properties and methods from traits

### PR DESCRIPTION
**Pull request by kukulich**
Fix of https://github.com/Roave/BetterReflection/issues/327.

### CODE REVIEW

File: test/unit/Fixture/ClassWithProperties.php [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<?php
```
Comment (Ocramius) [inline test/unit/Fixture/ClassWithProperties.php]: This file name does not reflect the intended usage - this should be `ClassWithPropertiesAndTraitProperties`

File: test/unit/Fixture/ClassWithMethods.php [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<?php
```
Comment (Ocramius) [inline test/unit/Fixture/ClassWithMethods.php]: This file name does not reflect the intended usage - this should be `ClassWithMethodsAndTraitMethods`

File: test/unit/Reflection/ReflectionPropertyTest.php [hunk 3]
```diff
@@ -362,4 +367,44 @@ public function testGetStartColumnAndEndColumn(string $php, int $startColumn, in
         self::assertEquals($startColumn, $constantReflection->getStartColumn());
         self::assertEquals($endColumn, $constantReflection->getEndColumn());
     }
+
+    public function testGetAst() : void
+    {
+        $php = '<?php
```
Comment (Ocramius) [inline test/unit/Reflection/ReflectionPropertyTest.php]: Can I suggest using `<<<'PHP'` here?

Comment (Ocramius) [reply test/unit/Reflection/ReflectionPropertyTest.php]: (maybe a sniff for anything above 3 `"\n"` in a string? Unsure how complex that would be)

**Comment by kukulich** [comment]
> Another test is needed

I'm not sure if I understand it well but I added another test.

File: test/unit/Fixture/ClassWithProperties.php (hunk 1 above)
Comment (kukulich) [reply test/unit/Fixture/ClassWithProperties.php]: Done.

File: test/unit/Fixture/ClassWithMethods.php (hunk 2 above)
Comment (kukulich) [reply test/unit/Fixture/ClassWithMethods.php]: Done.

File: test/unit/Reflection/ReflectionPropertyTest.php (hunk 3 above)
Comment (kukulich) [reply test/unit/Reflection/ReflectionPropertyTest.php]: Done.

**Comment by Ocramius** [comment]
@kukulich the tests you added are exactly what I meant. While they may sound obvious, they remove ambiguity around that API :+1:

**Comment by Ocramius** [comment]
:ship: \o/

**Pull request merged**